src/sov/mod.rs

```diff
@@ -0,0 +1,189 @@
+use std::{fs::read_to_string, path::Path, sync::Arc, time::Duration};
+
+use actix_web::rt;
+use cadence::{CountedExt, StatsdClient};
+use chrono::Utc;
+use config::ConfigError;
+use serde::{Deserialize, Serialize};
+use tokio::sync::RwLock;
+
+use crate::{
+    error::{HandlerError, HandlerErrorKind, HandlerResult},
+    settings::Settings,
+    web::middleware::sentry as l_sentry,
+};
+
+pub struct SOVManager {
+    pub refresh_rate: Duration,
+    pub last_response: Option<LastResponse>,
+    pub source_url: Option<url::Url>,
+}
+
+impl SOVManager {
+    pub async fn fetch(
+        &self,
+        storage_client: &cloud_storage::Client,
+    ) -> HandlerResult<Option<LastResponse>> {
+        if let Some(bucket) = &self.source_url {
+            let host = bucket
+                .host()
+                .ok_or_else(|| {
+                    HandlerError::internal(&format!("Missing bucket Host {:?}", self.source_url))
+                })?
+                .to_string();
+            let path = bucket.path().trim_start_matches('/');
+            let obj = storage_client.object().read(&host, path).await?;
+            if let Some(LastResponse { updated, .. }) = self.last_response {
+                // if the remote object is not newer than the local object, do nothing
+                if obj.updated <= updated {
+                    return Ok(None);
+                }
+            };
+
+            let bytes = storage_client.object().download(&host, path).await?;
+            let contents = String::from_utf8(bytes).map_err(|e| {
+                HandlerErrorKind::General(format!("Could not read SOV Settings: {:?}", e))
+            })?;
+            let new_response = serde_json::from_str(&contents).map_err(|e| {
+                HandlerErrorKind::General(format!("Could not read SOV Settings: {:?}", e))
+            })?;
+            return Ok(Some(LastResponse {
+                updated: obj.updated,
+                response: new_response,
+            }));
+        }
+        Ok(None)
+    }
+
+    pub fn update(&mut self, last_response: LastResponse) {
+        self.last_response = Some(last_response);
```

Comment:
We can also encode `last_response` (the json dump) here and store the encoded SOV so that the downstream consumers don't need to do that over and over again.

```rs
self.encoded_sov = SOME_BASE64_ENGINE::encode(serde_json::to_string(last_response.response));
```
Then `TilesHandlerResponse` can be:
```rs
pub struct TilesHandlerResponse {
    #[serde(flatten)]
    pub tile_response: adm::TileResponse,
    #[serde(
        rename = "sov",
        skip_serializing_if = " Option::is_none"
    )]
    pub sov_response: Option<String>,
}
```